nicegui/outbox.py

```diff
@@ -23,35 +23,46 @@ def __init__(self, client: Client) -> None:
         self.updates: Dict[ElementId, Optional[Element]] = {}
         self.messages: Deque[Message] = deque()
         self._should_stop = False
+        self._enqueue_event = None
         if core.app.is_started:
             background_tasks.create(self.loop(), name=f'outbox loop {client.id}')
         else:
             core.app.on_startup(self.loop)
 
+    def _set_enqueue_event(self):
+        """Set the enqueue event while accounting for lazy initialization."""
+        if self._enqueue_event:
+            self._enqueue_event.set()
+
     def enqueue_update(self, element: Element) -> None:
         """Enqueue an update for the given element."""
         self.updates[element.id] = element
+        self._set_enqueue_event()
 
     def enqueue_delete(self, element: Element) -> None:
         """Enqueue a deletion for the given element."""
         self.updates[element.id] = None
+        self._set_enqueue_event()
 
     def enqueue_message(self, message_type: MessageType, data: Any, target_id: ClientId) -> None:
         """Enqueue a message for the given client."""
         self.messages.append((target_id, message_type, data))
+        self._set_enqueue_event()
 
     async def loop(self) -> None:
         """Send updates and messages to all clients in an endless loop."""
+        self._enqueue_event = asyncio.Event()
+        self._enqueue_event.set()
+
         while not self._should_stop:
             try:
-                await asyncio.sleep(0.01)
-
-                if not self.updates and not self.messages:
-                    continue
-
+                await self._enqueue_event.wait()
+                await asyncio.sleep(0.005)
+                
                 if not self.client.has_socket_connection:
-                    continue
-
+                    await self.client.connected(timeout=2**30)
```

Comment:
I'm not sure if waiting basically indefinitely is a good idea. The client _might_ never connect, e.g. if the internet connection is interrupted before the websocket connection is created. I think we need to set the timeout to something like 60 seconds (like we do in `timer.py`) and `return` in case of a `TimeoutError`. This way we avoid a memory leak with broken client connections.

Reply:
I reasoned that if there's never a connection, we never have any work to do and that there's nothing we can do about that locally. The memory leak aspect didn't occur to me. With that in mind, I completely agree that it should be changed to 60 seconds.

Reply:
Changing that now.

Reply:
Actually, this will take me a bit. I want to be sure to get it right.

Reply:
I made the change to the connection timeout handling. I modeled it after the example you mentioned in timer.py.

Reply:
I'm having some second thoughts about this. Might it not be better to wait so we can recover if the connection is re-established. Otherwise, the UI would die and need to be reloaded because of an interruption lasting more than 60 seconds. I only ask because this is the current implementation, and we would be changing it.

Reply:
I was just looking through the code more and I realized a few things. To be consistent with the current behavior. We would need to occasionally time out `client.connected()` and make sure that `self._should_stop` has not been set. I also realized that the `self._enqueue_event.wait()` should time out periodically and do the same, so we don't get stuck waiting forever. It get set when the `client` calls `stop()`in its `delete()` method, so I think this would prevent any new memory leaks.

Reply:
I think with `await asyncio.wait_for(self._enqueue_event.wait(), timeout=1.0)` we basically reproduce the old behavior, just with a larger interval for checking `_should_stop`. Do you agree?

Reply:
Absolutely, I was about to do that myself.